ihmc-perception/src/main/java/us/ihmc/sensors/RealSensePublisher.java

```diff
@@ -0,0 +1,77 @@
+package us.ihmc.sensors;
+
+import perception_msgs.msg.dds.ImageMessage;
+import us.ihmc.commons.thread.ThreadTools;
+import us.ihmc.communication.CommunicationMode;
+import us.ihmc.communication.PerceptionAPI;
+import us.ihmc.communication.ROS2Tools;
+import us.ihmc.communication.ros2.ROS2DemandGraphNode;
+import us.ihmc.communication.ros2.ROS2Helper;
+import us.ihmc.euclid.referenceFrame.ReferenceFrame;
+import us.ihmc.perception.RawImage;
+import us.ihmc.perception.realsense.RealsenseConfiguration;
+import us.ihmc.perception.realsense.RealsenseDeviceManager;
+import us.ihmc.ros2.ROS2Node;
+import us.ihmc.ros2.ROS2Topic;
+import us.ihmc.tools.thread.ExecutorServiceTools;
+import us.ihmc.tools.thread.RestartableThread;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Supplier;
+
+public class RealSensePublisher
+{
+   private static final String REALSENSE_SERIAL_NUMBER = System.getProperty("d455.serial.number", "213522252883");
+   private static final ROS2Topic<ImageMessage> REALSENSE_COLOR_TOPIC = PerceptionAPI.D455_COLOR_IMAGE;
+   private static final ROS2Topic<ImageMessage> REALSENSE_DEPTH_TOPIC = PerceptionAPI.D455_DEPTH_IMAGE;
+
+   private final RealsenseColorDepthImageRetriever realsenseImageRetriever;
+   private final RealsenseColorDepthImagePublisher realsenseImagePublisher;
```

Comment:
The retriever and publisher both have `destroy` methods which should be called when the process ends. Perhaps a shutdown hook is the best way to call it? I can't think of a better method.